kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/SettingUpKolibri.vue

```diff
@@ -1,50 +1,133 @@
-            os_user: true,
+            on_my_own_setup: this.isOnMyOwnSetup,
+            os_user: this.canGetOsUser,
           },
+          auth_token: v4(),
         };
-        FacilityImportResource.createsuperuser(facilityUserData)
-          .then(
-            () => (window.location = window.urls['kolibri:kolibri.plugins.user_auth:user_auth']())
-          )
-          .catch(err => console.log(err));
+      },
+
+      /** Introspecting the machine via it's `state.context` properties */
+      isOnMyOwnSetup() {
+        return this.wizardService.state.context.onMyOwnOrGroup == UsePresets.ON_MY_OWN;
+      },
+      canGetOsUser() {
+        return this.wizardService.state.context.canGetOsUser;
+      },
+      isNewFacility() {
+        return this.wizardService.state.context.facilityNewOrImport === FacilityTypePresets.NEW;
+      },
+      isLearnOnlyDevice() {
+        return this.wizardService.state.context.fullOrLOD === DeviceTypePresets.LOD;
+      },
+    },
+    mounted() {
+      if (this.isOnMyOwnSetup) {
+        // 1) On my own
+        if (this.canGetOsUser) {
+          // 1.1) If we can get the OS user, we'll do this
+          this.createAndProvisionOnMyOwnUserApp();
+        } else {
+          // 1.2) If user is doing "on my own" setup but we cannot get OS user
+          this.createAndProvisionOnMyOwnUserDevice();
+        }
+      }
+
+      // From here, all are going to be "group learning" flows
+      if (!this.isLearnOnlyDevice) {
+        // 2) Group learning, Full device setup
+        if (this.isNewFacility) {
+          // 2.1) New Facility
+          this.createAndProvisionNewFullFacilityDevice();
+        } else {
+          // 2.2) Import Facility
+          // We already have the facility imported, just provision and redirect
+          this.provisionDevice();
+        }
       } else {
-        window.location = window.urls['kolibri:kolibri.plugins.user_auth:user_auth']();
+        // 3) Group learning, learn only device
+        this.provisionDevice();
       }
     },
     methods: {
-      isOnMyOwnSetup() {
-        return this.wizardService.state.context.onMyOwnOrGroup == UsePresets.ON_MY_OWN;
+      createAndProvisionOnMyOwnUserDevice() {
+        SetupWizardResource.createonmyownuser(this.facilityUserData).then(() =>
+          this.provisionDevice()
+        );
+      },
+      createAndProvisionOnMyOwnUserApp() {
+        SetupWizardResource.createappuser(this.facilityUserData).then(() => this.provisionDevice());
+      },
+      createAndProvisionNewFullFacilityDevice() {
+        SetupWizardResource.createsuperuser(this.facilityUserData).then(() =>
+          this.provisionDevice()
+        );
+      },
+      provisionDevice() {
+        SetupWizardResource.provisiondevice(this.deviceProvisioningData)
+          .then(() => {
+            // FIXME In dev mode, we'll wait 5 seconds before moving along so we can see the page
+            // ... maybe we keep this?
```

Comment:
I think waiting a few seconds might be nice, even if it is mostly like a "foaming toothpaste" effect. Maybe there is some UX research about this? sometimes realllyy short wait times feel glitchy to me.

Reply:
Thoughts @jtamiace ?

Reply:
And @tomiwaoLE ?

Reply:
What part of the onboarding flow is this? Having a hard time telling